### PR TITLE
plotter: Add FillColor field for BoxPlot and logic in plot

### LINKED ISSUE #674: plotter: Cannot set fill color on BoxPlot
### What are you trying to do?
Trying to set a FillColor on the BoxPlot struct after creating a new BoxPlot

### What did you expect to happen?
To be able to set a fill color on the boxplot (just like how Histogram works)

### What actually happened?
There seem's to be no field in the BoxPlot struct to set the color of the box 

### What version of Go and Gonum/plot are you using?
go version go1.15.6 darwin/amd64
gonum.org/v1/plot v0.8.1

### Does this issue reproduce with the current master?
Yes

### PR DESCRIPTION
**Pull request by rogerwelin**
This change adds FillColor field to the BoxPlot struct, initializes it as nil and adds non-nil logic in plot for both horizontal and vertical boxplots.

Closes #674

### CODE REVIEW

**Review by sbinet — COMMENTED** [review]
looks good.

could you also modify the box-plot examples to exercize this new feature?
(the reference plots for these examples would thus need to be updated)

**Comment by rogerwelin** [comment]
tests and golden pngs updated

**Review by sbinet — DISMISSED** [review]
LGTM.

last request: could you please send another PR against the [gonum/gonum](https://github.com/gonum/gonum) repo adding yourself to the `AUTHORS` and/or `CONTRIBUTORS` files?

thanks again.

**Comment by rogerwelin** [comment]
@sbinet Done! (https://github.com/gonum/gonum/pull/1572)
And thanks for helping me navigate the codebase

File: plotter/boxplot.go [hunk 1]
```diff
@@ -108,6 +113,7 @@ func NewBoxPlot(w vg.Length, loc float64, values Valuer) (*BoxPlot, error) {
 	b.CapWidth = 3 * w / 4
 
 	b.GlyphStyle = DefaultGlyphStyle
+	b.FillColor = nil
```
Comment (kortschak) [inline plotter/boxplot.go:116]: This line is not necessary; `b.FillColor` is nil on creation of `b`.

File: plotter/boxplot.go [hunk 2]
```diff
@@ -206,13 +212,19 @@ func (b *BoxPlot) Plot(c draw.Canvas, plt *plot.Plot) {
...
 		{X: x - b.Width/2 - b.BoxStyle.Width/2, Y: q1},
-	})
+	}
+
+	box := c.ClipLinesY(pts)
+
+	if b.FillColor != nil {
+		c.FillPolygon(b.FillColor, c.ClipPolygonY(pts))
+	}
 	c.StrokeLines(b.BoxStyle, box...)
```
Comment (kortschak) [inline plotter/boxplot.go:228]: I think this will read better as a single stanza, please remove the blank lines.

File: plotter/boxplot.go [hunk 3]
```diff
@@ -334,13 +346,20 @@ func (b horizBoxPlot) Plot(c draw.Canvas, plt *plot.Plot) {
...
-	})
+	}
+
+	box := c.ClipLinesX(pts)
+
+	if b.FillColor != nil {
+		c.FillPolygon(b.FillColor, c.ClipPolygonX(pts))
+	}
+
 	c.StrokeLines(b.BoxStyle, box...)
```
Comment (kortschak) [inline plotter/boxplot.go:363]: Same here.

File: plotter/boxplot_example_test.go [hunk 4]
```diff
@@ -34,14 +35,18 @@ func ExampleBoxPlot() {
...
 	normBox, err := plotter.NewBoxPlot(vg.Points(20), 1, normal)
+	normBox.FillColor = color.RGBA{127, 188, 165, 1}
 	if err != nil {
 		log.Panic(err)
 	}
 	expBox, err := plotter.NewBoxPlot(vg.Points(20), 2, expon)
+	expBox.FillColor = color.RGBA{127, 188, 165, 1}
 	if err != nil {
 		log.Panic(err)
 	}
```
Comment (kortschak) [inline plotter/boxplot_example_test.go:52]: The error checks should happen before use of value returned by `NewBoxPlot`.

**Review by kortschak — CHANGES_REQUESTED** [review]

**Comment by rogerwelin** [comment]
@kortschak PR updated with suggested fixes

File: plotter/boxplot.go [hunk 5]
```diff
@@ -64,6 +65,10 @@ type BoxPlot struct {
 	// GlyphStyle is the style of the outside point glyphs.
 	GlyphStyle draw.GlyphStyle
 
+	// FillColor is the color used to fill the box.
+	// Default value is nil
```
Comment (kortschak) [inline plotter/boxplot.go:69]: Sorry, I missed this. The default is nil by consequence of the language, so having a semantic doc would be better `// The default is no fill.` (also note doc sentences always have trailing periods).

Comment (rogerwelin) [reply plotter/boxplot.go]: Agreed. Updated

**Review by kortschak — APPROVED** [review]
Thanks

**Pull request merged**